### PR TITLE
update mathlib

### PR DESCRIPTION
**Pull request by enricozb**
updates mathlib, lean-toolchain, adds nix files, removes `.github`.

### CODE REVIEW

**Comment by RemyDegenne** [comment]
Thanks. Updating Mathlib is a good idea.
However there are changes here that I am not ready to merge:
- don't remove `.github`. It's the CI of the repository, that checks that the package builds and also generates the doc and the blueprint.
- In CLT.lean (renamed from Clt.lean), you commented out all the import lines and added an import Mathlib instead. Please revert that: the goal of that file is to import the content of the project.
- You added a bunch of flake/nix stuff. I don't know what that's for and would say that unless there is a good reason for them, they should not be added.
- You removed a lot of stuff from the lakefile. Why?

In summary, a Mathlib update is a good idea, but this PR does a lot of other more controversial things. Perhaps open an other PR with the update only?
Also if you want to work on the project, I can take the time to update it to use the new leanblueprint. This repository was created before leanblueprint was made easy to use, and it contains a lot of hacks copied from the PFR project.

**Comment by enricozb** [comment]
Whoops! Sorry I meant to make this PR to a private repo, was copying your project since it used lean blueprint.
